src/main/java/edu/byu/cs/controller/SubmissionController.java

```diff
@@ -558,29 +566,66 @@ public static void approveSubmission(
             LOGGER.warn("Approving submissions did not affect any submissions. Something probably went wrong.");
         }
 
-        // Determine approvedScore
-        if (approvedScore == null) {
-            float bestScoreForPhase = submissionDao.getBestScoreForPhase(studentNetId, phase);
-            if (bestScoreForPhase < 0.0f) {
-                throw new RuntimeException("Cannot determine best score for phase without any submissions in the phase.");
+        if (submissionToUse == null) {
+            submissionToUse = submissionDao.getBestSubmissionForPhase(studentNetId, phase);
+            if (submissionToUse == null) {
+                throw new RuntimeException("No submission was provided nor found for phase " + phase + " with user " + studentNetId);
             }
-            approvedScore = SubmissionHelper.prepareModifiedScore(bestScoreForPhase, scoreVerification);
         }
-        if (approvedScore <= 0.0f) {
-            throw new RuntimeException("Cannot set grade without a positive approvedScore!");
+        if (approvedScore == null) {
+            approvedScore = SubmissionHelper.prepareModifiedScore(submissionToUse.score(), penaltyPct);
         }
 
-        // Update grade-book
-        CanvasIntegration canvasIntegration = CanvasService.getCanvasIntegration();
-        // FIXME: Store `approvedScore` in the Grade-book
-        // canvasIntegration.submitGrade(studentNetId, approvedScore, );
-        if (true) {
-            throw new RuntimeException("ApproveSubmission not implemented!"); // TODO: Finish implementing method
+        float scoreDifference = approvedScore - originalScore;
+        RubricConfig rubricConfig = DaoService.getRubricConfigDao().getRubricConfig(phase);
```

Comment:
Were there extra rows I needed to add to my RubricConfig table? I didn't see any in the PR. As a result, the git commits rubric item is not being added to the canvas assessment, and no submission is receiving a penalty

Reply:
The git commits actually had been added in this PR before they were added to the `main` branch. Then those changes were added independently to the `main` branch. When the merge conflicts arose, Dallin kept the version already on `main`; this means that none of the lines will register here as additions because they already exist in the repo.

See these commits:
* ed8ecac20608b17b99a3a1589dd180c7d551704e (Authored first; #228 only)
* 1c3aa3a187d336d141581f9ec7247f70cbdd9d0c (Introduced in #342)

I'm not sure if you need to add anything your your `RubricConfig` table. The more likely reason this code isn't working is that it just hasn't been set up properly. I expect that I can look at the existing code from #233 which solves this problem and share the same code for use in this PR.

Reply:
Based on our discussion earlier, it sounds like we do need rows added to the table for this data.

I have no idea what data needs to be contained in the table, nor how to get it. @19mdavenport Can you please me generate the SQL statements that will insert the correct data?

Reply:
It's done. The SQL statement has been added to the PR description.